### PR TITLE
patch to graph groupchat

### PR DESCRIPTION
**Pull request by sonichi**
## Why are these changes needed?

Clean up a few remaining issues in graph group chat, such as the handling of `allow_repeat_speaker`, tests, bugs.

## Related issue number

#1541

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [x] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

File: autogen/agentchat/groupchat.py [hunk 1]
```diff
@@ -56,9 +66,7 @@ class GroupChat:
     admin_name: Optional[str] = "Admin"
     func_call_filter: Optional[bool] = True
     speaker_selection_method: Optional[str] = "auto"
-    allow_repeat_speaker: Optional[
-        Union[bool, List[Agent]]
-    ] = True  # It would be set to True if allowed_or_disallowed_speaker_transitions is None
+    allow_repeat_speaker: Optional[Union[bool, List[Agent]]] = None
```
[the remark this answers is not in the document]
Comment (joshkyh) [reply autogen/agentchat/groupchat.py:69]: My bad, I misread L82, sorry!

File: autogen/agentchat/groupchat.py [hunk 2]
```diff
@@ -42,12 +42,22 @@ class GroupChat:
...
-    - speaker_transitions_type: whether the speaker_transitions_type is a dictionary containing lists of allowed agents or disallowed agents. allowed means the allowed_or_disallowed_speaker_transitions is a dictionary containing lists of allowed agents. If set to disallowed, then the allowed_or_disallowed_speaker_transitions is a dictionary containing lists of disallowed agents. Must be supplied if allowed_or_disallowed_speaker_transitions is not None.
+    - allow_repeat_speaker: whether to allow the same speaker to speak consecutively.
+        Default is True, in which case all speakers are allowed to speak consecutively.
+        If `allow_repeat_speaker` is a list of Agents, then only those listed agents are allowed to repeat.
+        If set to False, then no speakers are allowed to repeat.
+        `allow_repeat_speaker` and `allowed_or_disallowed_speaker_transitions` are mutually exclusive.
+    - allowed_or_disallowed_speaker_transitions: a dictionary of keys and list as values.
+        The keys are the source agents, and the values are the agents that the key agent can transition to.
```
Comment (freedeaths) [inline autogen/agentchat/groupchat.py:51]: The keys are source agents, and the values are agents that the key agent can transit to.

Comment (sonichi) [reply autogen/agentchat/groupchat.py]: revised.

**Review by freedeaths — APPROVED** [review]
LGTM

**Pull request merged**